scripts/creation-portal-migration/template/generate/gProgram.js

```diff
@@ -0,0 +1,539 @@
+        // increment program nominated failed count and store the id
+        migratedCount.failed.program.nominated.count++;
+        if (!migratedCount.failed.program.nominated.ids.includes(id)) {
+          migratedCount.failed.program.nominated.ids.push(id);
+        }
+        logger.error(`Error while nominating program for solution_id: ${
+          solution?._id
+        } Error:
+      ${JSON.stringify(err?.response?.data)}`);
+      }
+    );
+
+    if (res.responseCode !== httpStatusCode.ok.code) {
+      // Update the database with migration reference if nomination fails
+      await updateSolutionDb(query, solution, migratedCount);
+      return;
+    }
+
+    logger.info(
+      `Sourcing Program nominated for solution_id: ${solution?._id} === ${programId}`
+    );
+
+    // update the query with isNominated
+    query = { ...query, [constants.MIGRATION_REFERENCE.IS_NOMINATED]: true };
+  } else {
+    // increment program nominated success count
+    migratedCount.success.program.existing.nominated++;
+  }
+
+  if (!solution?.migrationReference?.isContributorAdded) {
+    // Template to add the contributor
+    const addContributor = {
+      program_id: programId,
+      user_id:
+        userData?.srcOrgAdmin?.contributorOrgAdminId ||
+        process.env.DEFAULT_CONTRIBUTOR_ORG_ADMIN_ID,
+      rolemapping: {
+        CONTRIBUTOR: [
+          userData?.mappedUserId || process.env.DEFAULT_CONTRIBUTOR_USER_ID,
+        ],
+      },
+    };
+
+    // call the api to add contributor
+    const updateNomination = await updateContributorToProgram(
+      addContributor
+    ).catch((err) => {
+      // increment program contributor add failed count and store the id
+      migratedCount.failed.program.contributor.count++;
+      if (!migratedCount.failed.program.contributor.ids.includes(id)) {
+        migratedCount.failed.program.contributor.ids.push(id);
+      }
+
+      logger.error(`Error while adding contributor program for solution_id: ${
+        solution?._id
+      } Error:
+        ${JSON.stringify(err?.response?.data)}`);
+    });
+
+    if (updateNomination.responseCode !== httpStatusCode.ok.code) {
+      // Update the database if contributor update fails
+      await updateSolutionDb(query, solution, migratedCount);
+      return;
+    }
+
+    logger.info(
+      `Sourcing Program added contributor for solution_id: ${solution?._id} === ${programId}`
+    );
+
+    // update the query with isContributorAdded
+    query = {
+      ...query,
+      [constants.MIGRATION_REFERENCE.IS_CONTRIBUTOR_ADDED]: true,
+    };
+  } else {
+    // increment program contributor added success count
+    migratedCount.success.program.existing.contributor++;
+  }
+
+  if (!solution?.migrationReference?.isContributorAccepted) {
+    const accept_contributor = {
+      program_id: programId,
+      user_id:
+        userData?.srcOrgAdmin?.mappedUserId ||
+        process.env.DEFAULT_CONTRIBUTOR_ORG_ADMIN_ID,
+      status: "Approved",
```

Comment:
@jayesh9747 please follow rhe comment, please check and add all strings to common